datasource.go

```diff
@@ -0,0 +1,204 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/bitly/go-simplejson"
+	"github.com/grafana/grafana-plugin-model/go/datasource"
+	plugin "github.com/hashicorp/go-plugin"
+	"golang.org/x/net/context"
+	"golang.org/x/net/context/ctxhttp"
+)
+
+var httpClient = &http.Client{}
+
+type ClickhouseDatasource struct {
+	plugin.NetRPCUnsupportedPlugin
+}
+
+func (t *ClickhouseDatasource) Query(ctx context.Context, req *datasource.DatasourceRequest) (r *datasource.DatasourceResponse, err error) {
+	// catch all panics and override err return value
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("clickhouse plugin panicked: %#w", r)
+		}
+	}()
+
+	refId := req.Queries[0].RefId
+	modelJson, err := simplejson.NewJson([]byte(req.Queries[0].ModelJson))
+	if err != nil {
+		return nil, err
+	}
+
+	query := modelJson.Get("rawQuery").MustString()
```

Comment:
Shouldn't I let clickhouse handle this? It will return: 
![image](https://user-images.githubusercontent.com/5578150/81327317-ed85b380-904f-11ea-8ea1-cc4db823e51b.png)

Reply:
It would. My comment marked as `nit` and it is optional. Intention here was to fail fast with better error message, because we know for sure that empty query doesn't make sense.